test/component.spec.ts

```diff
@@ -1,13 +1,256 @@
-import { test } from "./components"
+import { Pool } from "pg"
+import SQL from "sql-template-strings"
+import { setTimeout } from "timers/promises"
+import { createPgComponent, IPgComponent } from "../src"
+import { logger, test } from "./components"
+import { mockConnect } from "./utils"
 
-test("subraph component", function ({ components, stubComponents }) {
+jest.mock("timers/promises")
+
+test("pg component", function ({ components }) {
   afterEach(() => {
     jest.resetAllMocks()
   })
 
+  describe("when starting a database", () => {
+    describe("and it's connected successfully", () => {
+      it("should call the connect method on the pool", async () => {
+        const { pg } = components
+        const pool = pg.getPool()
+
+        mockConnect(pool)
+        await pg.start()
+
+        expect(pool.connect).toHaveBeenCalledTimes(1)
+      })
+
+      it("should release the connect client", async () => {
+        const { pg } = components
+        const pool = pg.getPool()
+
+        const db = mockConnect(pool)
+        await pg.start()
+
+        expect(db.release).toHaveBeenCalledTimes(1)
+      })
+    })
+
+    describe("and the connect fails", () => {
+      it("should release the connect client", async () => {
+        const { pg } = components
+        const errorMessage = "Error message"
+        const pool = pg.getPool()
+
+        jest.spyOn(pool, "connect").mockImplementation(async () => {
+          throw new Error(errorMessage)
+        })
+
+        await expect(() => pg.start()).rejects.toThrow(errorMessage)
```

Comment:
If you remove the lambda, it should work as well, as the expect will take a promise and pass it through the rejects evaluation.
```suggestion
        await expect(pg.start()).rejects.toThrow(errorMessage)
```

Reply:
updated 😊